src/libs/validationHandler.ts

```diff
@@ -0,0 +1,88 @@
+import { NextFunction, Request, Response } from 'express';
+export default ( config ) => ( req: Request, res: Response, next: NextFunction  ) => {
+    const errors = [];
+    console.log( 'Inside ValidationHandler Middleware' );
+    console.log( req.body );
+    console.log( req.query );
+    console.log(Object.keys( req.query ).length );
+    const keys = Object.keys( config );
+    keys.forEach((key) => {
+      const obj = config[key];
+      console.log('key is' , key);
+      const values = obj.in.map( ( val ) => {
+          return req[ val ][ key ];
+      });
+
+      // Checking for In i.e Body or Query
+      console.log('body is', req[obj.in]);
+      console.log('body', Object.keys( req[obj.in] ).length );
+      if (Object.keys( req[obj.in] ).length === 0) {
+          errors.push({
+              message: `Values should be passed through ${obj.in}`,
+              status: 400
+          });
+      }
```

Comment:
This check is not correct. `Obj.in`  contains array not string if you have two values in array req[obj.in] will return undefined. Also checking the length of whole body or query object is not correct. You need to check the value for a particular input key/field